### PR TITLE
tldextract --update can be configured to download private domains

### LINKED ISSUE #64: tldextract --update does not download private domains
If I run tldextract --update to update domains file then, because of this line https://github.com/john-kurkowski/tldextract/blob/1.5.1/tldextract/tldextract.py#L264 the private domains are not downloaded.
Therefore, if later I create TLDExtract(include_psl_private_domains=True) class, the private domains are not mapped.
I recommend always downloading the full list and replace line 264 as 
TLD_EXTRACTOR = TLDExtract(include_psl_private_domains=True)

### PR DESCRIPTION
**Pull request by kristjanr**
fixes #64

### CODE REVIEW

File: tldextract/tldextract.py [hunk 1]
```diff
@@ -281,7 +281,7 @@ def _add_extra_suffixes(self, suffixes):
             return frozenset(suffixes)
         return suffixes
 
-TLD_EXTRACTOR = TLDExtract()
+TLD_EXTRACTOR = TLDExtract(include_psl_private_domains=True)
```
Comment (john-kurkowski) [inline tldextract/tldextract.py]: I like where this is going. I agree we must persist all public/private domains, but only include private results if the user opts into them. However this edit will include private domains for anybody who uses the export `extract`, just a few lines down. That would undo #19.

**Comment by kristjanr** [comment]
another try to fix #64

File: tldextract/tldextract.py [hunk 2]
```diff
@@ -374,12 +374,13 @@ def main():
     parser.add_argument('-c', '--cache_file', help='use an alternate TLD definition file')
 
     args = parser.parse_args()
+    tld_extract = TLDExtract(include_psl_private_domains=True)
```
Comment (john-kurkowski) [inline tldextract/tldextract.py]: Anybody who runs `tldextract --update` and later uses the same cache file will be forced to have public/private domains in their results, even if they only want public domains.

Ideally, we store metadata in the persisted file so results can be filtered at runtime.

For the 1.x series though, I think I'd be happy with `include_psl_private_domains` being configurable here.

Comment (john-kurkowski) [inline tldextract/tldextract.py]: For the ideal fix, I opened #66.

Comment (john-kurkowski) [inline tldextract/tldextract.py]: > For the 1.x series though, I think I'd be happy with include_psl_private_domains being configurable here.

Bump. Sorry if it wasn't clear, this is the only blocker on this PR. Should be an easy update.

(I don't think anybody's gonna tackle #66 anytime soon. Don't hold your breath for it.)

**Comment by kristjanr** [comment]
Huh?

**Comment by john-kurkowski** [comment]
If you make `include_psl_private_domains` configurable, defaulting to false, I can accept this. Otherwise, the PR will inflict private domains on everybody.

**Comment by john-kurkowski** [comment]
Thank you!

**Pull request merged**